pkg/gatewayserver/io/ws/ws_test.go

```diff
@@ -364,12 +374,16 @@ func TestDiscover(t *testing.T) {
 				t.Fatalf("Failed to write message: %v", err)
 			}
 
+			var wg sync.WaitGroup
+			wg.Add(1)
 			go func() {
-				_, _, err := conn.ReadMessage()
-				if err == nil {
```

Comment:
This assertion was actually wrong. `err` is supposed to be `nil` here, while the body (the 2nd parameter) will contain the error in JSON.